daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/iterable/SValueIterable.scala

```diff
@@ -12,7 +12,7 @@ private[speedy] object SValueIterable {
   that =>
   private[iterable] def iterator(v: SValue): Iterator[SValue] = v match {
     case SValue.SPAP(prim, actuals, _) => iterator(prim) ++ actuals.asScala.iterator
-    case SValue.SRecord(_, _, values) => values.asScala.iterator
+    case r: SValue.SRecordRep => r.dict.iterator.flatMap({ case (_, v) => Iterator(v) })
```

Comment:
Not sure how i feel about the inconsistency here between the other cases
minor style issue

Reply:
ok, we can change this!